criu/sk-unix.c

```diff
@@ -577,71 +504,60 @@ static int unix_resolve_name(int lfd, uint32_t id, struct unix_sk_desc *d,
 		if (get_mnt_id(lfd, &mnt_id))
 			return -1;
 		ue->mnt_id = mnt_id;
-		ue->has_mnt_id = mnt_id;
+		ue->has_mnt_id = true;
 	}
 
-	if (ue->mnt_id >= 0)
-		ns = lookup_nsid_by_mnt_id(ue->mnt_id);
-	else
-		ns = lookup_ns_by_id(root_item->ids->mnt_ns_id, &mnt_ns_desc);
-	if (!ns) {
-		ret = -ENOENT;
-		goto out;
+	fd = ioctl(lfd, SIOCUNIXFILE);
```

Comment:
SIOCUNIXFILE was added in 4.10. Should we care about older kernels?

Reply:
If `ioctl` fails, we could fallback to `resolve_rel_name`, and include a comment that `SIOCUNIXFILE` is not supported in older kernels.